src/callbacks.c

```diff
@@ -1,4 +1,12 @@
 #include "curl-common.h"
+#include "utils.h"
+
+int default_callback_progress(void *dummy,
+                              double dltotal, double dlnow,
+                              double ultotal, double ulnow)
+{
+  return !(R_ToplevelExec(check_interrupt_fn, NULL));
```

Comment:
OK we also need to test specifically on Windows. Luke Tierney has explained to me that the windows implementation of R's interruption mechanism is kinda hacky and a bit fragile.